src/network/networkd-ndisc.c

```diff
@@ -199,6 +206,13 @@ static int ndisc_request_route(Route *in, Link *link, sd_ndisc_router *rt) {
         if (route->mtu == 0)
                 route->mtu = mtu;
 
+        if (link->network->ipv6_hop_limit < 0 && hop_limit > 0) {
+                link->network->ipv6_hop_limit = hop_limit;
+                r = link_set_ipv6_hop_limit(link);
```

Comment:
I guess this should be set per route, instead of per link. No?

There may exist multiple routers and they may provide different limit. If we set the value to link, whole routes may be affected.

Reply:
yes updated and the router preference could change and they may have different hop limit